prog/dftb+/lib_dftbplus/parser.F90

```diff
@@ -264,6 +261,29 @@ subroutine parseHsdTree(hsdTree, input, parserFlags)
   end subroutine parseHsdTree
 
 
+  !> Converts input version to parser version and removes InputVersion node if present.
+  subroutine handleInputVersion(root, implicitParserVersion)
+
+    !> Root eventually containing InputVersion
+    type(fnode), pointer, intent(in) :: root
+
+    !> Parser version corresponding to input version, or unallocated if none has been found
+    integer, allocatable, intent(out) :: implicitParserVersion
+
+    type(fnode), pointer :: child, dummy
+    type(string) :: versionString
+
+    call getChild(root, "InputVersion", child, requested=.false.)
+    if (associated(child)) then
+      call getChildValue(child, "", versionString)
+      implicitParserVersion = parserVersionFromInputVersion(unquote(char(versionString)), child)
+      dummy => removeChild(root, child)
```

Comment:
If I understand `removeChild` correctly, `dummy` will be pointing to `child` after the child value is disconnected from the root. Wouldn't it be more intuitive to destroy the `dummy` node afterwards (even if `dummy` and `child` are pointing to the same thing)?

Reply:
I agree, that is didactically indeed more consistent.